### PR TITLE
Add new field selectors doc

### PR DESCRIPTION
**Pull request by lucperkins**
This PR addresses issue #8645 by adding documentation for kubectl field selectors, which are currently mentioned only in passing in the existing documentation.

### CODE REVIEW

File: content/en/docs/concepts/overview/working-with-objects/field-selectors.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+Field selectors can be used to select across multiple resource types. This `kubectl` command would select all Statefulsets *and* Services that are not in the `default` namespace:
+
+```shell
+$ kubectl get statefulsets,services --field-selector metadata.namespace!=default
+```
+
+## Non-existent field errors
+
+If you attempt to select for a field that doesn't exist for the specified resource(s), you will see an error like this:
```
Comment (cben) [inline content/en/docs/concepts/overview/working-with-objects/field-selectors.md]: It's not just whether the filed *exists*, IIUC most fields that exist are not usable in fieldSelector, only a few fields are supported, depending on object type?
E.g. The error message below lists only "metadata.name", "metadata.namespace" which are far from all ingress fields!

This was IMHO the main question in the issue that asked this to be documented.
- I'd mention that only some fields are supported up front at the start.
- Rename this section to "Supported fields" or "Non-supported fields" and change example to a field that exists but not supported for selection?

I don't know if maintaining a full list of supported fields in the docs is feasible.  If yes, probably not here but mark selectable fields in API docs?

Comment (lucperkins) [reply content/en/docs/concepts/overview/working-with-objects/field-selectors.md]: Unfortunately, maintaining a full list would likely be time prohibitive *unless* there were a way to autogenerate that list out of the Kubernetes codebase (and I doubt that that's currently possible).

**Comment by lucperkins** [comment]
@cben You are most welcome. Thanks for thorough critique! The doc benefited immensely from it!

File: content/en/docs/concepts/overview/working-with-objects/field-selectors.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+---
+title: Field Selectors
+weight: 60
+---
+
+_Field selectors_ enable you to [select Kubernetes resources](../kubernetes-objects) on the basis of the value of one or more resource fields. Here are some example field selector queries:
```
Comment (tengqm) [inline content/en/docs/concepts/overview/working-with-objects/field-selectors.md]: > "../kubernetes-objects"

I'm not sue if we should avoid relative paths in docs. The fact is that this is rare. Having relative paths in docs is building implicit links between topics. When one topic is later moved to other places in the directory, these relative references are difficult to find and to fix.

File: content/en/docs/concepts/overview/working-with-objects/field-selectors.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+$ kubectl get services --field-selector metadata.namespace!=default
+```
+
+{{< warning >}}
+Please note that *only* the `=` and `!=` operators are supported as field selector operators. [Label selector](../labels) syntax should *not* be taken as a general guide to field selector syntax. If you require more complex queries involving selectors like [`in` or `notin`](../labels#set-based-requirement), you may be better served by label selectors.
+{{< /warning >}}
+
+## Chained selectors
+
+As with [label](../labels) and other selectors, field selectors can be chained together as a comma-separated list. This `kubectl` command would select all Pods for which the `status.phase` does not equal `Running` *and* the `spec.restartPolicy` field equals `Always`:
```
Comment (tengqm) [inline content/en/docs/concepts/overview/working-with-objects/field-selectors.md]: > 'equal'

should be 'equal to'?

File: content/en/docs/concepts/overview/working-with-objects/field-selectors.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
+---
+title: Field Selectors
+weight: 60
+---
+
+_Field selectors_ enable you to [select Kubernetes resources](/docs/concepts/overview/working-with-objects/kubernetes-objects) on the basis of the value of one or more resource fields. Here are some example field selector queries:
```
Comment (zacharysarah) [inline content/en/docs/concepts/overview/working-with-objects/field-selectors.md]: From the [style guide](https://kubernetes.io/docs/home/contribute/style-guide/#content-best-practices): use simple and direct language.
```
_Field selectors_ let you [select Kubernetes resources](/docs/concepts/overview/working-with-objects/kubernetes-objects) based on the value of one or more resource fields. Here are some sample field selector queries:
```
So:
> _Field selectors_ let you [select Kubernetes resources](/docs/concepts/overview/working-with-objects/kubernetes-objects) based on the value of one or more resource fields. Here are some sample field selector queries:

File: content/en/docs/concepts/overview/working-with-objects/field-selectors.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+* `metadata.name=my-service`
+* `metadata.namespace!=default`
+* `status.phase=Pending`
+
+This `kubectl` command, for example, would select all Pods for which the value of the [`status.phase`](/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase) field is `Running`:
```
Comment (zacharysarah) [inline content/en/docs/concepts/overview/working-with-objects/field-selectors.md]:
```
This `kubectl` command selects all Pods for which the value of the [`status.phase`](/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase) field is `Running`:
```
So:
> This `kubectl` command selects all Pods for which the value of the [`status.phase`](/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase) field is `Running`:

File: content/en/docs/concepts/overview/working-with-objects/field-selectors.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+```shell
+$ kubectl get pods
+$ kubectl get pods --field-selector ""
+```
+{{< /note >}}
+
+## Supported fields
+
+Which fields can be used with field selectors varies by Kubernetes resource type. *All* resource types do, however, support the `metadata.name` and `metadata.namespace` fields. If you attempt to use field selectors on a field that isn't supported for that resource, you will receive an error. Here's an example:
```
Comment (zacharysarah) [inline content/en/docs/concepts/overview/working-with-objects/field-selectors.md]:
```
Supported field selectors vary by Kubernetes resource type. All resource types support the `metadata.name` and `metadata.namespace` fields. Using unsupported field selectors produces an error. For example:
```
So:
> Supported field selectors vary by Kubernetes resource type. All resource types support the `metadata.name` and `metadata.namespace` fields. Using unsupported field selectors produces an error. For example:

File: content/en/docs/concepts/overview/working-with-objects/field-selectors.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+```shell
+$ kubectl get ingress --field-selector foo.bar=baz
+Error from server (BadRequest): Unable to find "ingresses" that match label selector "", field selector "foo.bar=baz": "foo.bar" is not a known field selector: only "metadata.name", "metadata.namespace"
+```
+
+## Supported operators
+
+The `!=` field selector operator is supported in addition to `=` (used in the example directly above). This `kubectl` command, for example, would select all Kubernetes Services that are not in the `default` namespace:
```
Comment (zacharysarah) [inline content/en/docs/concepts/overview/working-with-objects/field-selectors.md]: Avoid passive voice when possible.
```
You can use the `!=` and `=` operators with field selectors. For example, this `kubectl` command selects all Kubernetes Services that aren't in the `default` namespace:
```
So:
> You can use the `!=` and `=` operators with field selectors. For example, this `kubectl` command selects all Kubernetes Services that aren't in the `default` namespace:

File: content/en/docs/concepts/overview/working-with-objects/field-selectors.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+```shell
+$ kubectl get services --field-selector metadata.namespace!=default
+```
+
+{{< warning >}}
+Please note that *only* the `=` and `!=` operators are supported as field selector operators. [Label selector](/docs/concepts/overview/working-with-objects/labels) syntax should *not* be taken as a general guide to field selector syntax. If you require more complex queries involving selectors like [`in` or `notin`](/docs/concepts/overview/working-with-objects/labels#set-based-requirement), you may be better served by label selectors.
```
Comment (zacharysarah) [inline content/en/docs/concepts/overview/working-with-objects/field-selectors.md]: Why is this a warning? What, per the style guide, makes this information crucial and/or dangerous? Specify the consequences.

I'll hold further feedback on this sentence until you've addressed the bit above. ☝️

File: content/en/docs/concepts/overview/working-with-objects/field-selectors.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+{{< /warning >}}
+
+## Chained selectors
+
+As with [label](/docs/concepts/overview/working-with-objects/labels) and other selectors, field selectors can be chained together as a comma-separated list. This `kubectl` command would select all Pods for which the `status.phase` does not equal `Running` *and* the `spec.restartPolicy` field equals `Always`:
```
Comment (zacharysarah) [inline content/en/docs/concepts/overview/working-with-objects/field-selectors.md]:
```
As with `[label](/docs/concepts/overview/working-with-objects/labels)` and other selectors, field selectors can be chained together as a comma-separated list. This `kubectl` command selects all Pods for which the `status.phase` does not equal `Running` and the `spec.restartPolicy` field equals `Always`:
```
So:
> As with `[label](/docs/concepts/overview/working-with-objects/labels)` and other selectors, field selectors can be chained together as a comma-separated list. This `kubectl` command selects all Pods for which the `status.phase` does not equal `Running` and the `spec.restartPolicy` field equals `Always`:

File: content/en/docs/concepts/overview/working-with-objects/field-selectors.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+```shell
+$ kubectl get pods --field-selector=status.phase!=Running,spec.restartPolicy=Always
+```
+
+## Multiple resource types
+
+Field selectors can be used to select across multiple resource types. This `kubectl` command would select all Statefulsets *and* Services that are not in the `default` namespace:
```
Comment (zacharysarah) [inline content/en/docs/concepts/overview/working-with-objects/field-selectors.md]: Avoid the passive voice.
```
You can use field selectors across multiple resource types. This `kubectl` command selects all Statefulsets and Services that are not in the `default` namespace:
```
So:
> You can use field selectors across multiple resource types. This `kubectl` command selects all Statefulsets and Services that are not in the `default` namespace:

**Comment by lucperkins** [comment]
@zacharysarah Okay, I've made the requested wording changes and also decided to simply remove the warning that was included previously.

**Comment by indradhanush** [comment]
Thanks for adding the docs. The document states that:
> Supported field selectors vary by Kubernetes resource type

Is there a way to find out if a specific field is supported in the field selectors without trying to use it first? I ran across a use case where I thought I could use them in my code but it turns out that the field is not supported.

**Comment by lucperkins** [comment]
@indradhanush I've looked through the Kubernetes codebase and unfortunately I don't think that there's a way to determine that systematically.

**Comment by indradhanush** [comment]
@lucperkins I see. Thanks for looking though. :+1:

File: content/en/docs/concepts/overview/working-with-objects/field-selectors.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+Supported field selectors vary by Kubernetes resource type. All resource types support the `metadata.name` and `metadata.namespace` fields. Using unsupported field selectors produces an error. For example:
+
+```shell
+$ kubectl get ingress --field-selector foo.bar=baz
+Error from server (BadRequest): Unable to find "ingresses" that match label selector "", field selector "foo.bar=baz": "foo.bar" is not a known field selector: only "metadata.name", "metadata.namespace"
+```
+
+## Supported operators
+
+You can use the `!=` and `=` operators with field selectors. This `kubectl` command, for example, selects all Kubernetes Services that aren't in the `default` namespace:
```
Comment (yue9944882) [inline content/en/docs/concepts/overview/working-with-objects/field-selectors.md]: @lucperkins Should we also document usage of "==" too?😃

Comment (lucperkins) [reply content/en/docs/concepts/overview/working-with-objects/field-selectors.md]: Good catch! Fixed

**Pull request merged**